src/plugins/discover/public/application/helpers/get_sharing_data.ts

```diff
@@ -20,13 +20,9 @@ export async function getSharingData(
   currentSearchSource: ISearchSource,
   state: AppState | SavedSearch,
   config: IUiSettingsClient
-) {
+): Promise<{ columns: string[]; searchSource: SearchSourceFields }> {
```

Comment:
What do you think about moving the type into a separate interface instead of inlining? That can help in the future when we need to update that.
```
interface SharingData {
  columns: string[];
  searchSource: SearchSourceFields;
}
```

```suggestion
): Promise<SharingData> {
```